include/drivers/dai.h

```diff
@@ -0,0 +1,717 @@
+#define DAI_OPT_BIT_CLK_CONSUMER               BIT(1)
+/** DAI driver is frame clock provider (codec is consumer) */
+#define DAI_OPT_FRAME_CLK_PROVIDER            (0 << 2)
+/** DAI driver is frame clock consumer (codec is provider) */
+#define DAI_OPT_FRAME_CLK_CONSUMER             BIT(2)
+
+/** @brief Loop back mode.
+ *
+ * In loop back mode RX input will be connected internally to TX output.
+ * This is used primarily for testing.
+ */
+#define DAI_OPT_LOOPBACK                    BIT(7)
+
+/** @brief Ping pong mode
+ *
+ * Inherited from legacy TBD to check.
+ *
+ * In ping pong mode TX output will keep alternating between a ping buffer
+ * and a pong buffer. This is normally used in audio streams when one buffer
+ * is being populated while the other is being played (DMAed) and vice versa.
+ * So, in this mode, 2 sets of buffers fixed in size are used. Static Arrays
+ * are used to achieve this and hence they are never freed.
+ */
+#define DAI_OPT_PINGPONG                    BIT(6)
+
+/** \brief Types of DAI
+ *
+ * The type of the DAI. This ID type is used to configure bespoke DAI HW
+ * settings.
+ *
+ * DAIs have a lot of physical link feature variability and therefore need
+ * different configuration data to cater for different use cases. We
+ * sometimes may need to pass extra configuration data prior to DAI start.
+ */
+enum dai_type {
+	DAI_LEGACY_I2S = 0,	/**< Legacy I2S compatible with i2s.h */
+	DAI_INTEL_SSP,		/**< Intel SSP */
+	DAI_INTEL_DMIC,		/**< Intel DMIC */
+	DAI_INTEL_HDA,		/**< Intel HD/A */
+	DAI_INTEL_ALH,		/**< Intel ALH */
+	DAI_IMX_SAI,                /**< i.MX SAI */
+	DAI_IMX_ESAI,               /**< i.MX ESAI */
+	DAI_AMD_BT,			/**< Amd BT */
+	DAI_AMD_SP,			/**< Amd SP */
+	DAI_AMD_DMIC,		/**< Amd DMIC */
+	DAI_MEDIATEK_AFE,            /**< Mtk AFE */
+	DAI_INTEL_SSP_NHLT,            /**< nhlt ssp */
+	DAI_INTEL_DMIC_NHLT,            /**< nhlt ssp */
+	DAI_INTEL_HDA_NHLT,		/**< nhlt Intel HD/A */
+	DAI_INTEL_ALH_NHLT,		/**< nhlt Intel ALH */
+};
+
+/**
+ * @brief I2C Direction
+ */
+enum dai_dir {
+	/** Receive data */
+	DAI_DIR_RX = 0,
+	/** Transmit data */
+	DAI_DIR_TX,
+	/** Both receive and transmit data */
+	DAI_DIR_BOTH,
+};
+
+/** Interface state */
+enum dai_state {
+	/** @brief The interface is not ready.
+	 *
+	 * The interface was initialized but is not yet ready to receive /
+	 * transmit data. Call dai_configure() to configure interface and change
+	 * its state to READY.
+	 */
+	DAI_STATE_NOT_READY = 0,
+	/** The interface is ready to receive / transmit data. */
+	DAI_STATE_READY,
+	/** The interface is receiving / transmitting data. */
+	DAI_STATE_RUNNING,
+	/** The interface is clocking but not receiving / transmitting data. */
+	DAI_STATE_PRE_RUNNING,
+	/** The interface is draining its transmit queue. */
+	DAI_STATE_STOPPING,
+	/** TX buffer underrun or RX buffer overrun has occurred. */
+	DAI_STATE_ERROR,
+	/** TBD not used today, here for completeness */
+	DAI_STATE_DRAINING,
```

Comment:
Looks like we can combine STOPPING with DRAINING states (based on the comments from the Zephyr I2S code)

Reply:
ok